x/compliance/handler_test.go

```diff
@@ -342,6 +342,258 @@ func TestHandler_ProvisionDifferentModels(t *testing.T) {
+	_, err := setup.Handler(setup.Ctx, certifyModelMsg)
+	require.NoError(t, err)
+
+	// query certified model
+	receivedComplianceInfo, _ := queryComplianceInfo(setup, vid, pid, softwareVersion, dclcompltypes.ZigbeeCertificationType)
+	checkCertifiedModelInfo(t, certifyModelMsg, receivedComplianceInfo)
+
+	originalComplianceInfo, _ := queryComplianceInfo(setup, vid, pid, softwareVersion, dclcompltypes.ZigbeeCertificationType)
+	originalDeviceSoftwareCompliance, _ := queryDeviceSoftwareCompliance(setup, receivedComplianceInfo.CDCertificateId)
+
+	updateComplianceInfoMsg := &types.MsgUpdateComplianceInfo{
+		Creator:           setup.CertificationCenter.String(),
+		Vid:               vid,
+		Pid:               pid,
+		SoftwareVersion:   softwareVersion,
+		CertificationType: dclcompltypes.ZigbeeCertificationType,
+		CDCertificateId:   "new_cd_certificate_id",
+	}
+
+	_, err = setup.Handler(setup.Ctx, updateComplianceInfoMsg)
+	require.NoError(t, err)
+
+	updatedComplianceInfo, _ := queryComplianceInfo(setup, vid, pid, softwareVersion, dclcompltypes.ZigbeeCertificationType)
+
+	require.Equal(t, "new_cd_certificate_id", updatedComplianceInfo.CDCertificateId)
+
+	require.Equal(t, updatedComplianceInfo.CDVersionNumber, originalComplianceInfo.CDVersionNumber)
+	require.Equal(t, updatedComplianceInfo.Date, originalComplianceInfo.Date)
+	require.Equal(t, updatedComplianceInfo.Reason, originalComplianceInfo.Reason)
+	require.Equal(t, updatedComplianceInfo.CertificationRoute, originalComplianceInfo.CertificationRoute)
+	require.Equal(t, updatedComplianceInfo.ProgramType, originalComplianceInfo.ProgramType)
+	require.Equal(t, updatedComplianceInfo.ProgramTypeVersion, originalComplianceInfo.ProgramTypeVersion)
+	require.Equal(t, updatedComplianceInfo.CompliantPlatformUsed, originalComplianceInfo.CompliantPlatformUsed)
+	require.Equal(t, updatedComplianceInfo.CompliantPlatformVersion, originalComplianceInfo.CompliantPlatformVersion)
+	require.Equal(t, updatedComplianceInfo.Transport, originalComplianceInfo.Transport)
+	require.Equal(t, updatedComplianceInfo.FamilyId, originalComplianceInfo.FamilyId)
+	require.Equal(t, updatedComplianceInfo.SupportedClusters, originalComplianceInfo.SupportedClusters)
+	require.Equal(t, updatedComplianceInfo.OSVersion, originalComplianceInfo.OSVersion)
+	require.Equal(t, updatedComplianceInfo.ParentChild, originalComplianceInfo.ParentChild)
+	require.Equal(t, updatedComplianceInfo.CertificationIdOfSoftwareComponent, originalComplianceInfo.CertificationIdOfSoftwareComponent)
+
+	deviceSoftwareCompliance, err := queryDeviceSoftwareCompliance(setup, updateComplianceInfoMsg.CDCertificateId)
+	require.NoError(t, err)
+
+	require.Equal(t, len(deviceSoftwareCompliance.ComplianceInfo), len(originalDeviceSoftwareCompliance.ComplianceInfo))
+
+	for _, softwareCompliance := range deviceSoftwareCompliance.ComplianceInfo {
+		require.Equal(t, softwareCompliance.CDCertificateId, updateComplianceInfoMsg.CDCertificateId)
```

Comment:
Please check that queryDeviceSoftwareCompliance by old CDCertificateId returns nothing.